settlement-pipelines/src/reporting.rs

```diff
@@ -98,45 +325,78 @@ impl ErrorHandler {
             }
             Err(err) => {
                 for single_error in err.into_iter() {
-                    self.add_retry_able_error(anyhow::Error::from(single_error));
+                    self.retryable()
+                        .with_msg(format!("{}", single_error))
+                        .with_source("transaction_execution")
+                        .add();
                 }
             }
         }
     }
 
+    pub fn get_warnings(&self) -> Vec<&ErrorEntry> {
+        self.entries.iter().filter(|e| e.is_warning()).collect()
+    }
+
+    pub fn get_errors(&self) -> Vec<&ErrorEntry> {
+        self.entries.iter().filter(|e| e.is_critical()).collect()
+    }
+
+    pub fn get_retryable_errors(&self) -> Vec<&ErrorEntry> {
+        self.entries.iter().filter(|e| e.is_retryable()).collect()
+    }
+
+    pub fn get_infos(&self) -> Vec<&ErrorEntry> {
+        self.entries
+            .iter()
+            .filter(|e| e.severity() == ErrorSeverity::Info)
```

Comment:
(non-blocking)
It would make sense to me to create `is_info` in `ErrorEntry` (~same implementation like for other severity levels) and use it here, so it would be consistent with other `get_*` functions above.